x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/index.ts

```diff
@@ -244,6 +251,17 @@ interface FieldBasedOperationDefinition<C extends BaseIndexPatternColumn> {
    * together with the agg configs returned from other columns.
    */
   toEsAggsConfig: (column: C, columnId: string, indexPattern: IndexPattern) => unknown;
+  /**
+   * Validate that the operation has the right preconditions in the state. For example:
+   *
+   * - Requires a date histogram operation somewhere before it in order
+   * - Missing references
+   */
+  getErrorMessage: (
```

Comment:
Do we need this here? It seems like the `getErrorMessage` on `BaseOperationDefinitionProps` makes sure it will be around.

Reply:
I wanted to make it obligatory for Field based operations (it's optional for `BaseOperationDefinitionProps`)

Reply:
Gotcha, makes sense.